packages/lix-sdk/src/database/applySchema.ts

```diff
@@ -134,6 +136,46 @@ export async function applySchema(args: { sqlite: SqliteDatabase }) {
 
   INSERT OR IGNORE INTO label (name) VALUES ('confirmed');
 
+  -- branches
+
+  CREATE TABLE IF NOT EXISTS branch (
+    id TEXT PRIMARY KEY DEFAULT (uuid_v4()),
+
+    -- name is optional. 
+    -- 
+    -- "anonymous" branches can ease workflows. 
+    -- For example, a user can create a branch 
+    -- without a name to experiment with
+    -- changes with no mental overhead of 
+    -- naming the branch.
+    name TEXT
+  
+  ) strict;
+
+  CREATE TABLE IF NOT EXISTS branch_change_pointer (
+    branch_id TEXT NOT NULL,
+    change_id TEXT NOT NULL,
+    change_file_id TEXT NOT NULL,
+    change_entity_id TEXT NOT NULL,
+    change_type TEXT NOT NULL,
+
+    PRIMARY KEY(branch_id, change_file_id, change_entity_id, change_type),
+
+    FOREIGN KEY(branch_id) REFERENCES branch(id),
+    FOREIGN KEY(change_id, change_file_id, change_entity_id, change_type) REFERENCES change(change_id, change_file_id, change_entity_id, change_type)
+  ) strict;
```

Comment:
Decided to not go with the change set column to 

- get foreign key validations 
- avoid "dead" pointers e.g. a change set can have no reference. a change pointer of a branch with no reference can be garbage collected
- indexes 👀 

An open question is whether `branch_change_head` is a better suited table name